druid/src/test/java/org/apache/calcite/test/DruidAdapterIT.java

```diff
@@ -1695,28 +1694,120 @@ public Void apply(ResultSet resultSet) {
-        + "'intervals':['1900-01-09T00:00:00.000/2992-01-10T00:00:00.000'],"
-        + "'filter':{'type':'and',"
-        + "'fields':[{'type':'bound','dimension':'cases_per_pallet','lower':'8',"
-        + "'lowerStrict':false,'ordering':'numeric'},"
-        + "{'type':'bound','dimension':'cases_per_pallet','upper':'10','upperStrict':false,"
-        + "'ordering':'numeric'},{'type':'bound','dimension':'units_per_case','upper':'15',"
-        + "'upperStrict':true,'ordering':'numeric'}]},'dimensions':[],'metrics':['store_sales'],"
-        + "'granularity':'all','pagingSpec':{'threshold':16384,'fromNext':true},"
-        + "'context':{'druid.query.fetch':false}}";
+    String druidQuery = "{\"queryType\":\"timeseries\",\"dataSource\":\"foodmart\","
+        + "\"descending\":false,\"granularity\":\"all\",\"filter\":{\"type\":\"and\","
+        + "\"fields\":[{\"type\":\"bound\",\"dimension\":\"cases_per_pallet\",\"lower\":\"8\","
+        + "\"lowerStrict\":false,\"ordering\":\"numeric\"},{\"type\":\"bound\","
+        + "\"dimension\":\"cases_per_pallet\",\"upper\":\"10\",\"upperStrict\":false,"
+        + "\"ordering\":\"numeric\"},{\"type\":\"bound\",\"dimension\":\"units_per_case\","
+        + "\"upper\":\"15\",\"upperStrict\":true,\"ordering\":\"numeric\"},"
+        + "{\"type\":\"selector\",\"dimension\":\"__time\",\"value\":\"1997\","
+        + "\"extractionFn\":{\"type\":\"timeFormat\",\"format\":\"yyyy\",\"timeZone\":\"UTC\","
+        + "\"locale\":\"en-US\"}}]},\"aggregations\":[{\"type\":\"doubleSum\","
+        + "\"name\":\"EXPR$0\",\"fieldName\":\"store_sales\"}],"
+        + "\"intervals\":[\"1900-01-09T00:00:00.000/2992-01-10T00:00:00.000\"],"
+        + "\"context\":{\"skipEmptyBuckets\":true}}";
     sql(sql)
         .queryContains(druidChecker(druidQuery))
-        .explainContains("PLAN=EnumerableInterpreter\n"
-             + "  BindableAggregate(group=[{}], EXPR$0=[SUM($1)])\n"
-             + "    BindableFilter(condition=[AND(>=(/INT(Reinterpret($0), 86400000), 1997-01-01), "
-             + "<(/INT(Reinterpret($0), 86400000), 1998-01-01))])\n"
-             + "      DruidQuery(table=[[foodmart, foodmart]], "
-             + "intervals=[[1900-01-09T00:00:00.000/2992-01-10T00:00:00.000]], "
-             + "filter=[AND(>=(CAST($11):BIGINT, 8), <=(CAST($11):BIGINT, 10), "
-             + "<(CAST($10):BIGINT, 15))], projects=[[$0, $90]])\n")
         .returnsUnordered("EXPR$0=75364.09998679161");
   }
+
+  @Test public void testPushOfFilterExtractionOnDayAndMonth() {
+    String sql = "SELECT \"product_id\" , EXTRACT(day from \"timestamp\"), EXTRACT(month from "
+        + "\"timestamp\") from \"foodmart\" WHERE  EXTRACT(day from \"timestamp\") >= 30 AND "
+        + "EXTRACT(month from \"timestamp\") = 11 "
+        + "AND  \"product_id\" >= 1549 group by \"product_id\", EXTRACT(day from "
+        + "\"timestamp\"), EXTRACT(month from \"timestamp\")";
+    sql(sql)
```

Comment:
We should add the DruidQuery in this case too, even if we verify that we are actually pushing the EXTRACT for other queries below.